tools/Normalise_Consent_Data.php

```diff
@@ -0,0 +1,307 @@
+            array_push($errors, "The date of consent is missing for " . $consentName . ":
+                       [ID]     => " . $consentID . "
+                       [CandID] => " . $candID . "
+                       All 'yes' statuses must have a consent date.");
+        }
+        // Check if consent status is no and consent date given, but withdrawal date is empty
+        if($status === "no" && !empty($date)) {
+            if(empty($withdrawal)) {
+                array_push($errors, "The date of withdrawal is missing for " . $consentName . ":
+                           [ID]     => " . $consentID . "
+                           [CandID] => " . $candID . "
+                           All 'no' statuses with given consent date must have a withdrawal date.");
+            }
+        }
+        // Check for zero dates
+        if($date === "0000-00-00" || $withdrawal === "0000-00-00") {
+            array_push($errors, "Zero dates found in participant_status for " . $consentName . ":
+                       [ID]     => " . $consentID . "
+                       [CandID] => " . $candID . "
+                       Please remove date or run /tools/DB_date_zeros_removal.php.");
+        }
+    }
+}
+////////////////////////////////////////////////
+// CHECK NEW TABLES AND CONFIG SETTINGS EXIST //
+////////////////////////////////////////////////
+
+// Check useConsent was inserted into ConfigSettings
+try {
+    $configID = $db->pselectOne("SELECT ID FROM ConfigSettings WHERE Name='useConsent'", array());
+} catch (DatabaseException $e) {
+    array_push($errors, "useConsent missing from ConfigSettings table.
+               Run SQL/New_patches/2018-03-01_normalise_consent.sql");
+}
+try {
+    $configValue = $db->pselectOne("SELECT Value FROM Config WHERE ConfigID=:configid",
+                       array('configid'=>$configID));
+} catch (DatabaseException $e) {
+    array_push($errors, "useConsent's Config value missing from Config table.
+               Run SQL/New_patches/2018-03-01_normalise_consent.sql");
+}
+if (!$db->tableExists('consent') || !$db->tableExists('candidate_consent_rel') || !$db->tableExists('candidate_consent_history')) {
+    array_push($errors, "New, normalized consent tables do not exist.
+               Run SQL/New_patches/2018-03-01_normalise_consent.sql");
+}
+//////////////////
+// THROW ERRORS //
+//////////////////
+if (!empty($errors)) {
+    print_r($errors);
+    echo "\nResolve errors and run script again.\n";
+    die();
+} else {
+    echo "\nValidation successful.\n";
+}
+/////////////////////
+// CONTINUE SCRIPT //
+/////////////////////
+
+// Update ConfigSetting table with value of 'useConsent' if true. Default is set to false.
+if ($useConsent === "true") {
+    $updateValue = ['Value' => $useConsent];
+    $db->update(
+        'Config',
+        $updateValue,
+        array(
+         'ConfigID' => $configID, 
+        )
+    );
+    echo "\nConfig settings set to use consent.\n\n";
```

Comment:
What happens if useConsent isn't "true"?

Reply:
if it is not true, consent isn't being used, and so tables are normalized and data imported as normal but config setting stays false (the default set from ConfigTables.sql and patch). this will eventually be used in the config module to render or hide the 'Consent Status' tab in Candidate parameters.

Reply:
https://github.com/aces/Loris/pull/3560/files moves config from config.xml to DB